### ISSUE TITLE
Incorrect handling qrexec policy rules

### ISSUE THREAD

**Issue by unman** [QubesOS/qubes-issues#8227]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### Qubes OS release
4.1

### Brief summary
qrexec policy rules are not handled correctly, with effect that it is possible to qvm-copy to a prohibited qube.

### Steps to reproduce
Create rules in /etc/qubes/policy.d/30-user.policy:
```
qubes.Filecopy  *  @anyvm  vault   deny
qubes.Filecopy  *  foo     @anyvm  allow target=vault
```
Attempt to qvm-copy file in foo

### Expected behavior
Filecopy will be blocked by 1st rule

### Actual behavior
File is copied to vault, although precedent rule explicitly blocks transfers.

**Comment by andrewdavidwong** [comment]
I can also reproduce this.

**Comment by ben-grande** [comment]
PR submitted :)

**Comment by marmarek** [comment]
> Diagnosis: Redirect (`target=` parameter, but also `default_target=`) does not check against a list of targets that were not denied, instead it redirects to any target that is available in your system.

This is intentional for `target=` - see https://www.qubes-os.org/doc/qrexec/#policy-files:
> Note that if the request is redirected (target= parameter), policy action remains the same – even if there is another rule which would otherwise deny such request.

But `default_target` should not have this side effect, it's only about pre-selecting target from otherwise allowed ones.

**Comment by ben-grande** [comment]
> This is intentional for target= - see https://www.qubes-os.org/doc/qrexec/#policy-files:
>> Note that if the request is redirected (target= parameter), policy action remains the same – even if there is another rule which would otherwise deny such request.

I understand it is intentional. It is okay to override the `destination` on the same line by using `target=`, but is it okay to ignore previous deny rules to that `destination`?

My PR aboves consider previous deny rules and if encounters none, uses the redirect `target=`.

Shouldn't this intentional behavior change? When using Admin Management VMs or qubes 90-*.policy, should their rule (later sourced/read) always override the user rule (previous sourced/read)?

**Comment by marmarek** [comment]
Duplicate of https://github.com/QubesOS/qubes-issues/issues/7723
